cli/command/service/update.go

```diff
@@ -185,6 +179,13 @@ func runUpdate(dockerCli *command.DockerCli, flags *pflag.FlagSet, serviceID str
 		updateOpts.RegistryAuthFrom = types.RegistryAuthFromSpec
 	}
 
+	// TODO(nishanttotla): Figure out correct registry auth to pass here
```

Comment:
`updateOpts.EncodedRegistryAuth` seems right. This will be set if `--with-registry-auth` is specified. There probably isn't much point in using auth credentials to resolve the digest if those credentials won't be passed on with the service for pulling.